keps/sig-network/1880-multiple-service-cidrs/README.md

```diff
@@ -0,0 +1,954 @@
+  <https://github.com/kubernetes/kubernetes/issues/87603>
+- the size of the Service Cluster CIDR, for IPv4 the prefix size is limited to /12, however, for
+  IPv6 it is limited to /112 or fewer. This restriction is causing issues for IPv6 users, since /64
+  is the standard and minimum recommended prefix length
+- only is possible to use reserved a range of Service IP addresses using the feature gate
+  "keps/sig-network/3070-reserved-service-ip-range"
+  <https://github.com/kubernetes/kubernetes/issues/95570>
+
+### Goals
+
+Implement a new allocation logic for Services IPs that:
+
+- scales well with the number of reservations
+- the number of allocation is tunable
+- is completely backwards compatible
+
+### Non-Goals
+
+- Any change unrelated to Services
+- Any generalization of the API model that can evolve onto something different, like an IPAM API, or
+  collide with existing APIs like Ingress and GatewayAPI.
+- NodePorts use the same allocation model than ClusterIPs, but it is totally out of scope of this
+  KEP. However, this KEP can be a proof that a similar model will work for NodePorts too.
+- Change the default IPFamily used for Services IP allocation, the defaulting depends on the
+  services.spec.IPFamily and services.spec.IPFamilyPolicy, a simple webhook or an admission plugin
+  can set this fields to the desired default, so the allocation logic doesn't have to handle it.
+- Removing the apiserver flags that define the service IP CIDRs, though that may be possible in the future.
+
+## Proposal
+
+The proposal is to implement a new allocator logic that uses 2 new API Objects: ServiceCIDRConfig
+and IPAddress, and allow users to dynamically increase the number of Services IPs available by
+creating new ServiceCIDRConfigs.
+
+The new allocator will be able to "automagically" consume IPs from any ServiceCIDRConfig available, we can
+think about this model, as the same as adding more disks to a Storage system to increase the
+capacity.
+
+To simplify the model, make it backwards compatible and to avoid that it can evolve into something
+different and collide with other APIs, like Gateway APIs, we are adding the following constraints:
+
+- a ServiceCIDRConfig will be immutable after creation (to be revisited before Beta).
+- a ServiceCIDRConfig can only be deleted if there are no Service IP associated to it (enforced by finalizer).
+- there can be overlapping ServiceCIDRConfigs.
+- the apiserver will periodically ensure that a "default" ServiceCIDRConfig exists to cover the service CIDR flags
+  and the "kubernetes.default" Service.
+- any IPAddress existing in a cluster has to belong to a ServiceCIDRConfig.
+- any Service with a ClusterIP assigned is expected to have always a IPAddress object associated.
+- a ServiceCIDRConfig which is being deleted can not be used to allocate new IPs
+
+ This creates a 1-to-1 relation between Service and IPAddress, and a 1-to-N relation between
```

Comment:
Should these be `ServiceIPAddress` objects then? Rather than generic `IPAddress`?

Reply:
it started like that, but seems @thockin foresee a possible interaction with Gateway ... actually, if you see the API scheme, the reference to the parent object is generic and not tied to Services

Reply:
Probably - I am on the fence.  I'd like to think this same range mechanism COULD be used by Gateways (as a more modern clusterip) to allocate IPs